[cloudflare] Add documentation for new tag cache options (#99)
opennextjs/docs: pages/cloudflare/caching.mdx

#### On-Demand Revalidation

The tag revalidation mechanism can use either a [Cloudflare D1](https://developers.cloudflare.com/d1/) database or [Durable Objects](https://developers.cloudflare.com/durable-objects/) with SqliteStorage as its backing store for information about tags, paths, and revalidation times.

To use on-demand revalidation, you should also follow the [ISR setup steps](#incremental-static-regeneration-isr).

<Callout>
  If your app **only** uses the pages router, it does not need to have a tag cache and should skip this step.

>   You can also skip this step if you don't want to use `revalidateTag` or `revalidatePath` in your app.

Reviewer: ```suggestion
  You can also skip this step if your app doesn't to use `revalidateTag` nor `revalidatePath`.
```

</Callout>

> There is 3 different options to choose from for the tag cache: 'd1NextTagCache', `ShardedTagCache` and `d1TagCache`.

Reviewer: ```suggestion
There are 3 different options to choose from for the tag cache: 'd1NextTagCache', `ShardedTagCache` and `d1TagCache`.
```

> For a small to medium size project, `d1NextModeTagCache` is recommended, for bigger projects `ShardedTagCache` is recommended.

Reviewer: Everyone think their project is bigger :)
Are there guidelines for users to evaluate the size?

Author: It's not easy, it depends on the load that D1 is able to handle sustainably (Not sure about this), and also on the app itself.
If you use `revalidateTag` a lot, it will put a lot of additional strain on D1 that an app that barely use it won't (for the exact same load)

Reviewer: "There is" -> There are

Reviewer: Show we mention usage of `revalidateTag` instead of project size then?
